snapcraft/plugins/cmake.py

```diff
@@ -64,22 +64,15 @@ def __init__(self, name, options, project):
         self.build_packages.append('cmake')
 
     def build(self):
-        if os.path.exists(self.builddir):
-            shutil.rmtree(self.builddir)
-        os.mkdir(self.builddir)
-
-        source_subdir = getattr(self.options, 'source_subdir', None)
-        if source_subdir:
-            sourcedir = os.path.join(self.sourcedir, source_subdir)
```

Comment:
This is not needed, because the base class sets builddir to self.build_basedir + self.source_subdir. It is safe to just use self.builddir.